### ISSUE TITLE
babel.config.mjs not working

### ISSUE THREAD

**Issue by kotarofuma** [babel/babel#11958]
**Current behavior**
On node@14.8.0, I have 2 equivalent simple files: babel.config.cjs and babel.config.mjs (except for the exporting keyword)...

```js
const presets = [
  [
    "@babel/env",
    {
      targets: {
        edge: "10",
        firefox: "50",
        chrome: "60",
        safari: "11.1",
      },
      useBuiltIns: "usage",
      corejs: 2,
    }
  ]
];

module.exports = { presets };
```

commenting the mjs, everything works perfect doing npx babel src/index.js, while commenting the cjs one (enabling esm support), I get 
"Error: /Users/kotaro/Projects/ro/test/babel.config.mjs: Error while loading config - Cannot find module 'file:///Users/kotaro/Projects/ro/test/babel.config.mjs'
Require stack:
- /Users/kotaro/Projects/ro/test/node_modules/@babel/core/lib/config/files/import.js
- /Users/kotaro/Projects/ro/test/node_modules/@babel/core/lib/config/files/module-types.js
- /Users/kotaro/Projects/ro/test/node_modules/@babel/core/lib/config/files/configuration.js
- /Users/kotaro/Projects/ro/test/node_modules/@babel/core/lib/config/files/index.js
- /Users/kotaro/Projects/ro/test/node_modules/@babel/core/lib/index.js
- /Users/kotaro/Projects/ro/test/node_modules/@babel/cli/lib/babel/options.js
- /Users/kotaro/Projects/ro/test/node_modules/@babel/cli/lib/babel/index.js
- /Users/kotaro/Projects/ro/test/node_modules/@babel/cli/bin/babel.js"

Tracking the sources, it seems you guys are transpiling import into a require at https://github.com/babel/babel/blob/main/packages/babel-core/src/config/files/import.js:

```js
function import_(filepath) {
  console.log('000000', filepath);
  return Promise.resolve(`${filepath}`).then(s => _interopRequireWildcard(require(s)));
}
```
so that filepath is an absolute URL: { href: 'file:///Users/kotaro/Projects/ro/test/babel.config.mjs', ...} which seems to fail loading (yes, file is there).

Setting type: "module" in package json provokes same error

**Expected behavior**
babel should handle equivalently cjs and mjs if node version supports it

**Environment**
node@14.8.0

```
- Babel version(s): 7.11.1
- Node/npm version: [e.g. Node 14.8.0/npm 6.14.7]
- OS: OSX 10.13.4
- Monorepo: no
- How you are using Babel: cli ATM

**Possible Solution**


**Additional context**
Add any other context about the problem here. Or a screenshot if applicable

**Comment by JLHwung** [comment]
This issue is a regression. Affected versions: 7.11.0, 7.11.1.

**Comment by JLHwung** [comment]
Fixed in `@babel/core@7.11.4`.
